tools/hilow.wdl

```diff
@@ -0,0 +1,468 @@
+version 1.1
+
+task active_regions_merge {
+    meta {
+        description: "Merge active regions"
+        outputs: {
+            combined_promoters: "Merged active regions"
+        }
+    }
+
+    parameter_meta {
+        promoters: "Promoter regions in BED format"
+        loop_bed: "Loop anchors in BED format"
+        bam: "BAM file"
+    }
+
+    input {
+        File promoters
+        File loop_bed
+        File bam
+    }
+
+    String outfile = basename(promoters, ".bed") + ".LoopAnchors.Enhs.combined.sort.bed"
+
+    #@ except: LineWidth
+    command <<<
+        prom=~{basename(promoters,".bed")}
+        sort \
+            -k1,1 \
+            -k2,2n \
+            -k4,4 \
+            ~{promoters} | \
+            awk \
+            -F'\t' \
+            -v OFS="\t" \
+            '{ a[$1"\t"$2"\t"$3]=($1"\t"$2"\t"$3 in a? a[$1"\t"$2"\t"$3]",":"")$4 }END{ for(i in a) print i,a[i] }' | \
+            sort \
+            -k1,1 \
+            -k2,2n \
+            > $prom".unique.bed"
+
+        #get read counts
+        intersectBed -c -a $prom".unique.bed" -b ~{bam} > $prom"_Signal"
+
+        #compute top 2/3
+        length=$(wc -l $prom"_Signal" | awk -F\  '{print $1}')
+        keepN=$((length * 2 / 3))
+
+        sort -k5,5nr $prom"_Signal" \
+            | head -n "$keepN" \
+            | sort -k1,1 -k2,2n \
+            | cut -f 1-3 \
+            > "${prom}_Signal.keep.${keepN}.3cols.bed"
+
+        #get non-promoter region
+        cut -f 1-3 ~{loop_bed} | \
+            sort -k1,1 -k2,2n \
+            > ~{basename(loop_bed)}
+        bedtools \
+            subtract \
+            -a ~{basename(loop_bed)} \
+            -b $prom".unique.bed" | \
+            sort \
+            > Enhancers.Subtract.4kbPromoters.sort.bed
+
+        cat \
+            "${prom}_Signal.keep.${keepN}.3cols.bed" \
+            Enhancers.Subtract.4kbPromoters.sort.bed \
+            | sort -k1,1 -k2,2n \
+            > ~{outfile}
+    >>>
+
+    output {
+        File combined_promoters = outfile
+    }
+
+    runtime {
+        cpu: 8
```

Comment:
This should follow the standard `npcu` conventions. But I don't see what operations here benefit from more than one cpu. 8 is a lot of juice, is that a sane default?